infra/Pos/Slotting/Impl/Util.hs

```diff
@@ -17,55 +17,79 @@ import           Pos.Core.Types              (EpochIndex, SlotId (..), Timestamp
                                               mkLocalSlotIndex)
 import           Pos.Util.Util               (leftToPanic)
 
+import           Pos.Slotting.Util           (getEpochSlottingDataEmpatically)
 import           Pos.Slotting.MemState.Class (MonadSlotsData (..))
-import           Pos.Slotting.Types          (EpochSlottingData (..), SlottingData (..))
+import           Pos.Slotting.Types          (EpochSlottingData (..))
 
 -- | Approximate current slot using outdated slotting data.
-approxSlotUsingOutdated :: MonadSlotsData m => EpochIndex -> Timestamp -> m SlotId
+approxSlotUsingOutdated
+    :: (MonadSlotsData m)
+    => EpochIndex
+    -> Timestamp
+    -> m SlotId
 approxSlotUsingOutdated penult t = do
```

Comment:
This function still uses `penult` as the name of argument. And there are more usages of this word in your branch. I think it makes sense to remove all of them because you changed the naming:

```
gromak ~/Work/cardano-sl (ksaric/CSM-337|…) % scripts/grep.sh -i "penult"                                                                                                                                                                                                 0 :)
test/Test/Pos/Block/Logic/Mode.hs:                                                 waitPenultEpochEqualsDefault)
test/Test/Pos/Block/Logic/Mode.hs:    waitPenultEpochEquals = waitPenultEpochEqualsDefault
test/Test/Pos/Block/Logic/Mode.hs:    waitPenultEpochEquals = waitPenultEpochEqualsDefault
update/Pos/Update/Poll/Logic/Base.hs:            ("can't update slotting data, stored penult epoch is "%int%
update/Pos/Update/DB.hs:    esdPenult :: EpochSlottingData
update/Pos/Update/DB.hs:    esdPenult = EpochSlottingData
update/Pos/Update/DB.hs:    genesisSlottingData = createInitSlottingData esdPenult esdLast
infra/Pos/Slotting/MemState/Holder.hs:        penultEpoch <- getCurrentEpochIndex <$> readTVar var
infra/Pos/Slotting/MemState/Holder.hs:        when (penultEpoch /= target) retry
infra/Pos/Slotting/Impl/Util.hs:approxSlotUsingOutdated penult t = do
infra/Pos/Slotting/Impl/Util.hs:        if | t < epochStart -> SlotId (penult + 1) minBound
infra/Pos/Slotting/Impl/Util.hs:           | otherwise      -> outdatedEpoch systemStart t (penult + 1) sdLast
lwallet/Pos/Wallet/Light/Mode.hs:    waitPenultEpochEquals = error "notImplemented"
```